crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -81,37 +81,62 @@ async fn app_validation_workflow_inner(
 ) -> WorkflowResult<WorkComplete> {
     let env = workspace.dht_env.clone().into();
     let sorted_ops = validation_query::get_ops_to_app_validate(&env).await?;
-    tracing::debug!("app validating {} ops", sorted_ops.len());
+    let start_len = sorted_ops.len();
+    tracing::debug!("validating {} ops", start_len);
+    let start = (start_len >= NUM_CONCURRENT_OPS).then(std::time::Instant::now);
+    let saturated = start.is_some();
 
     // Validate all the ops
-    let iter = sorted_ops.into_iter().map(|so| {
+    let iter = sorted_ops.into_iter().map({
         let network = network.clone();
-        let conductor_handle = conductor_handle.clone();
         let workspace = workspace.clone();
-        let dna_hash = dna_hash.clone();
-        async move {
-            let (op, op_hash) = so.into_inner();
-            let author = op.header().author().clone();
-            let op_light = op.to_light();
-
-            // Validate this op
-            let r = validate_op(dna_hash, op, &conductor_handle, &(*workspace), &network).await;
-            (op_hash, author, op_light, r)
+        move |so| {
+            let network = network.clone();
+            let conductor_handle = conductor_handle.clone();
+            let workspace = workspace.clone();
+            let dna_hash = dna_hash.clone();
+            async move {
+                let (op, op_hash) = so.into_inner();
+                let dependency = get_dependency(op.get_type(), &op.header());
+                let op_light = op.to_light();
+
+                // Validate this op
+                let r = validate_op(dna_hash, op, &conductor_handle, &(*workspace), &network).await;
+                (op_hash, dependency, op_light, r)
+            }
         }
     });
     use futures::stream::StreamExt;
     let mut iter = futures::stream::iter(iter)
         .buffer_unordered(NUM_CONCURRENT_OPS)
-        .ready_chunks(NUM_CONCURRENT_OPS);
+        .ready_chunks(NUM_CONCURRENT_OPS * 100);
+    let (tx, rx) = tokio::sync::mpsc::channel(NUM_CONCURRENT_OPS * 100);
+    let jh = tokio::spawn(async move {
+        while let Some(op) = iter.next().await {
+            if tx.send(op).await.is_err() {
+                tracing::warn!("app validation task has failed to send ops. This is not a problem if the conductor is shutting down");
+                break;
+            }
+        }
+    });
```

Comment:
I don't understand this, why spawn a task to produce items on a channel that immediately get consumed? Is it for some chunking behavior that can't be achieved some other way? Can you add a comment?